Infrastructure/Infrastructure-SDLC.md

```diff
@@ -0,0 +1,186 @@
+Agile Manifest provide main value and principles.
+for example values:
+- Individuals and interactions over processes and tools
+- Working software over comprehensive documentation
+- Customer collaboration over contract negotiation
+- Responding to change over following a plan
+
+---
+
+## Question 6
+### Scrum Cycle HIGH
+It called sprint 1-2week but not more than 6 week.
+
+---
+
+## Question 7
+### SCRUM Estimation HIGH
+Each team member estimate task
+
+Story Point / Planning Poker - used with cards. The numbers are usually in the Fibonacci sequence: 0, 1, 2, 3, 5, 8, 13, and 21.
+
+T-Shirt Sizes - used with T-shirt sizes: extra small, small, medium, large, extra-large, etc.
+
+Large/Uncertain/Small
+
+The team chooses the easiest task and assigns it the minimum value. for example one story point. Time for implementing in hours is different for different team member,  junior/middle/senior developer.
+
+---
+
+## Question 8
+### How do you estimate a task? HIGH
+* I break the task into small tasks. A small problem can be estimated more precisely.
+* I estimate in story points all such small tasks.
+
+---
+
+## Question 9
+###  Testing Pyramid HIGH
+User Interface (UI) tests are at the top of the Testing Pyramid because they involve the largest scope and are the most complex. UI tests check all the system layers from UI to databases and external services. (more complex)
+
+Integration tests validate the interaction of a piece of code with external components. For this reason, they run slower than unit tests.
+
+Unit tests are at the base of the Testing Pyramid because they take the least time and cost. They check the logic of a small function or class. If a unit test fails, it means there are problems with the code.
+
+---
+
+## Question 10
+### Unit Tests HIGH
+1. Focus on one specific piece of the system in isolation
+2. Are easier to write, faster to execute, and cheaper to maintain
+3. Have no external dependency―any external dependency is mocked
+4. Let you know the exact piece of code where the error is
+
+---
+
+## Question 11
+### Design Patterns. Definition HIGH
+Design patterns are reusable solutions to commonly occurring problems in software design.
+
+---
+
+## Question 12
+### Benefits of Patterns HIGH
+* Patterns are proven solutions
+* Patterns can be easily reused
+* Patterns can be expressive(voc for developers)
+---
+
+## Question 13
+### Creation Patterns HIGH
+This pattern is provides various object creation mechanisms which increase flexibility and reuse of existing code
+* Builder - constructs complex objects by separating construction and representation
+* Factory method - creates objects without specifying the exact class to create
+* Singleton - restricts object creation for a class to only one instance
+
+---
+
+## Question 14
+### Behavioral Patterns HIGH
+Behavioral design patterns are design patterns that identify common communication patterns among objects.
+
+---
+
+## Question 15
+### SOLID Principles HIGH
```

Comment:
In addition to naming SOLID principles, consider adding real examples from JS world for each term